### ISSUE TITLE
sort order of contacts strange/unusable

### ISSUE THREAD

**Issue by zoff99** [SMSSecure/SMSSecure#266]
if you want to send a new message to a friend you select the "pen" button
then enter in the "Enter name or number" field:
"fr"

now you would expect a contact named "frank" to be shown on top of the list.
but not so. many contacts are shown, many of them don't have the letters "fr" in their name.
and frank is NOT listed on top.

this is very unusable. please make that filtering/matching so that nomal people can use/understand it

**Comment by BastienLQ** [comment]
I can't reproduce the bug. Can you attach a screenshot and a debug log?

**Comment by zoff99** [comment]
![1_resize](https://cloud.githubusercontent.com/assets/6833516/10560548/b23049fa-750e-11e5-8953-63a7e2847d27.png)
![2_resize](https://cloud.githubusercontent.com/assets/6833516/10560549/b48dd366-750e-11e5-941e-350af06d4cb9.png)
![3_resize](https://cloud.githubusercontent.com/assets/6833516/10560550/b8d796e6-750e-11e5-921b-254f433d04af.png)
![4_resize](https://cloud.githubusercontent.com/assets/6833516/10560552/bcbe1ac8-750e-11e5-8430-bf4d575afded.png)

**Comment by zoff99** [comment]
(just to be clear those phone numbers are random numbers and not real! dont call them.)

1) enter "FP" and it displays frank ? totally wrong
2) enter "F" and it displays "DUBH" as first result. this does not match at all -> totally wrong
3) enter "X" it displays "FRED" this does not match at all -> totally wrong
4) enter "GA" it displays "FRANK" this does not match at all -> totally wrong

that is just not usable by any normal human!

**Comment by johanw666** [comment]
I think it has to do something with the predictive text input: you are sending "fp" to the search list, and because no contact starts with "fp" it makes this up.

**Comment by zoff99** [comment]
thats a very disturbing comment.
please fix the problem. it's clearly documented

**Comment by sevan6** [comment]
@zoff99 
I tried to reproduce the behaviour you describe.
In the main screen I use the 'pen' icon in the bottom left.
When typing 'f' I have the contacts including f that are  listed
When followed by a p to get 'fp' in the search field, I have no contact listed (whereas I have a "Franck" in my contacts.

I use AnySoftKeyboard as a keyboard, I tried the AOSP (stock) keyboard and did not reproduce the behaviour neither.

To sum up : when typing in a sequence that is not in my contact names, the display is empty, it does not propose any contact.

To be sure, can you try another input method (another keyboard)?

**Comment by pR0Ps** [comment]
@zoff99 I'm also not seeing the issue you're describing. When I search it matches from the start of any name - first, last, or middle (ex: "s", "su", "sup", "m", "mc", "mca" would all match "Super McAwesome". It also searches by initials (ex: "sm" would also match "Super McAwesome"). Anything other than those methods of searching returned back no results.

I haven't looked at the searching code in a while so I'm not 100% on what it's doing but it shouldn't be anything crazy that would give the results you're seeing (as you say, that would be a bug).

To make sure it's not your keyboard entering random predictions could you try with the AOSP keyboard and/or the Google keyboard and report back?

**Comment by pR0Ps** [comment]
This has gone unreproduced without a follow-up for 2+ months. 

@zoff99: Feel free to repoen if you can still reproduce this bug in the latest version.
